File: services/worker-python/app/logger.py
import json
import time
import sys

def log(level, message, **fields):
    entry = {
        "timestamp": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
        "level": level,
        "service": "worker-python",
        "message": message,
        **fields,
    }
    sys.stdout.write(json.dumps(entry) + "\n")
    sys.stdout.flush()

File: services/core-go/internal/logger/logger.go
package logger

import (
	"encoding/json"
	"log"
	"time"
)

func Info(message string, fields map[string]string) {
	entry := map[string]string{
		"level":     "info",
		"service":   "core-go",
		"timestamp": time.Now().Format(time.RFC3339),
		"message":   message,
	}

	for k, v := range fields {
		entry[k] = v
	}

	b, _ := json.Marshal(entry)
	log.Println(string(b))
}

